parse/node.go

```diff
@@ -52,18 +52,33 @@ func (n *Node) Validate() error {
 		return fmt.Errorf("%s: no keys", n.Pos())
 
 	case 1:
+		if n.IsDefault() {
+			break
+		}
 		return fmt.Errorf("%s: missing name", n.Pos())
 
 	case 2:
 		if n.IsModule() {
 			return fmt.Errorf("%s: missing source or name in module call", n.Pos())
 		}
 
+		if n.IsDefault() {
+			return fmt.Errorf("%s: too many keys", n.Pos())
+		}
+
+		if n.IsCase() {
+			return fmt.Errorf("%s: missing name or predicate in case", n.Pos())
+		}
+
 	default:
 		if n.IsModule() && len(n.Keys) == 3 {
 			break
 		}
 
+		if n.IsCase() && len(n.Keys) == 3 {
+			break
+		}
+
```

Comment:
yeah, I was going to change and it and realized it was a bit too tangential to justify as part of this change set.  I'll add an issue.